### DISCUSSION TITLE
is it possible to use database other than couchdb? like aws dynamodb for example

### DISCUSSION THREAD

**Discussion by kurakurakantor** [vrtmrz/obsidian-livesync#60] [Q&A] [answered]
first of all, i just want to say that this is the coolest and most useful plugin i've come across. thank you so much for making this! after figuring out how to sync my obsidian for free for days, i came across this and this is definitely the best solution. unfortunately i can't create a free account on ibm cloudant, starting from october last year, we need credit card to register even a free accouned, and my virtual credit card won't be accepted by them. i was trying to look for alternatives but can't really find another free couchdb host like ibm, i came across some other ones but they're not couchdb, i wonder if that could still work? i'm also trying to figure all these out, experimenting, i have no knowledge about databases whatsoever so i'm sorry if this is a dumb question. once again thank you very much for the amazing plugin!

**Comment by vrtmrz** [comment, marked as answer]
Thank you for making this discussion! I think this is very suitable to think with everybody.

>  i came across this and this is definitely the best solution.

I'm very honored to hear that!

But unfortunately, there is no other practicable backend now. 

Let me explain.

CouchDB is a particular database that is very good at synchronization, and LiveSync relies on that powerful mechanism. With other databases, synchronization would have been harder. (especially with over three devices).

PouchDB - CouchDB compatible local database - can use a custom backend with a custom adapter. But almost adapters are not stable and are not maintained.
Like [this](https://github.com/search?o=desc&p=1&q=pouchdb+adapter&s=updated&type=Repositories)

As you said, There was no CouchDB hosting except IBM Cloudant now. I'm very disappointed too.
But if you are troubled with making accounts of IBM Cloudant, and you can use dynamodb (Does this mean having an AWS account?): you can serve your CouchDB in your AWS EC2 or ECS.

If you think, "I may host my server": We can choose some cloud infrastructure. I'm using Oracle Cloud Infrastructure for free.

**Reply by kurakurakantor** [reply]
thank you so much for the reply! and about the ec2 with couchdb, i actually just finished setting up my first successful attempt at that! im a total beginner so i had no idea how i was supposed to do any of it but after experimenting and a ton of research, it's all working now! i didn't even need to setup a https domain for my android. i'm so glad because i learned a lot along the way haha and now i just have to find a more permanent solution since ec2 on aws is only free for a year. oh i've been eyeing on oracle cloud too! aws was the only one i could access because it's the only one that accepts virtual credit card, but i've heard oracle cloud accepts debit card too so maybe that could work :D. thanks again for your answer! this will definitely help confused people like me to be able to use this really nice syncing solution!
